testing readings #962 #C20

KySuongLam opened the pull request:
# Description

Added C20 test case: "28 day shift end 2022-10-31 17:12:34 (partial hour) for 15 minute reading intervals and quantity units & kWh as kWh". 
Contributer: @selenasat @KySuongLam

Partially Addresses testing readings #962 

## Type of change

- [ ] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist
- [x] I have followed the [OED pull request](https://openenergydashboard.org/developer/pr/) ideas
- [x] I have removed text in ( ) from the issue request
- [x] You acknowledge that every person contributing to this work has signed the [OED Contributing License Agreement](https://openenergydashboard.org/developer/cla/) and each author is listed in the Description section.

## Limitations

selenasat committed:
added c20

KySuongLam committed:
First Update

KySuongLam committed:
Merge branch 'development' of https://github.com/selenasat/OED into development

selenasat committed:
Add unit and conversion data for Electric to kW in readings API tests

selenasat committed:
Merge branch 'development' of https://github.com/selenasat/OED into development

selenasat committed:
Add API request for comparing meter readings with date range and unit ID

KySuongLam committed:
adding expectCompare

selenasat committed:
No code changes

selenasat committed:
Merge branch 'development' of https://github.com/selenasat/OED into development

selenasat committed:
Fix unit ID in readings comparison test from 'kW' to 'kWh'

KySuongLam committed:
Fix Error

KySuongLam committed:
Merge branch 'development' of https://github.com/selenasat/OED into development

selenasat committed:
Fix indentation in readings comparison test for better readability

selenasat committed:
Merge branch 'development' of https://github.com/selenasat/OED into development

selenasat committed:
Fix unit ID in readings comparison test from 'kW' to 'kWh' and clean up test structure

selenasat committed:
Update expected value in readings comparison test for accuracy

KySuongLam committed:
Clean the code comment

KySuongLam committed:
Merge branch 'development' of https://github.com/selenasat/OED into development

Chocopepero commented:
Thank you for your contribution to OED, before we get started can I ask you do the following:
1. @KySuongLam, please sign the CLA. @selenasat signed it back in Feburary 2024.
2. In the Pull Request itself, please change "Fixes #962" to "Partially Addresses #962"

KySuongLam commented:
@Chocopepero I have already edited the Pull Request and signed the CLA. Please help us check again. Thank you!

Chocopepero reviewed (approved):
Tests run as expected and everything looks good.

huss committed:
fix description mistake

I put kWh instead of kW for the description in
the design doc. This fixes that mistake for the PR.
It was not a mistake by the developers but me.

huss reviewed (approved):
The review by @Chocopepero is taken as approving. Congratulations to @selenasat & @KySuongLam on their first accepted contribution to OED.
